src/mlpack/methods/kernel_svm/kernel_svm_impl.hpp

```diff
@@ -0,0 +1,247 @@
+        while (j == i)
+        {
+          j = rand() % data.n_cols;
+        }
+        E(j) = b + arma::sum(alpha.t() % label % K.col(j).t()) - label(j);
+
+        double alpha_i_old = alpha(i);
+        double alpha_j_old = alpha(j);
+
+        if (label(i) == label(j))
+        {
+          L = std::max(0.0, alpha(j) + alpha(i) - C);
+          H = std::min(C, alpha(j) + alpha(i));
+        }
+        else
+        {
+          L = std::max(0.0, alpha(j) - alpha(i));
+          H = std::min(C, C + alpha(j) - alpha(i));
+        }
+
+        if (L == H)
+          continue;
+
+        eta = 2 * K(i, j) - K(i, i) - K(j, j);
+        if (eta >= 0)
+          continue;
+
+        alpha(j) = alpha(j) - (label(j) * (E(i) - E(j))) / eta;
+
+        alpha(j) = std::min(H, alpha(j));
+        alpha(j) = std::max(L, alpha(j));
+
+        if (std::abs(alpha(j) - alpha_j_old) < tol)
+        {
+          alpha(j) = alpha_j_old;
+          continue;
+        }
+
+        alpha(i) = alpha(i) + label(i) * label(j) * (alpha_j_old - alpha(j));
+
+        double b1 = b - E(i)
+                    - label(i) * (alpha(i) - alpha_i_old) *  K(i, j)
+                    - label(j) * (alpha(j) - alpha_j_old) *  K(i, j);
+        double b2 = b - E(j)
+                    - label(i) * (alpha(i) - alpha_i_old) *  K(i, j)
+                    - label(j) * (alpha(j) - alpha_j_old) *  K(j, j);
+
+        if (0 < alpha(i) && alpha(i) < C)
+          b = b1;
+        else if (0 < alpha(j) && alpha(j) < C)
+          b = b2;
+        else
+          b = (b1 + b2) / 2;
+
+        num_changed_alphas = num_changed_alphas + 1;
+      }
+    }
+
+    if (num_changed_alphas == 0)
+      count = count + 1;
+    else
+      count = 0;
+  }
+  parameters = (data * (alpha.t() % label).t()).t();
+
+  y = arma::zeros(1 , data.n_cols);
+  x = arma::zeros(data.n_rows, data.n_cols);
+  for (size_t i = 0; i < data.n_cols; i++)
+  {
+    if (alpha(i) > 0)
+    {
+      y(i) = 1;
+      x.col(i) = data.col(i);
+    }
+  }
+}
+
+template <typename MatType, typename KernelType>
+void KernelSVM<MatType, KernelType>::Classify(
+    const MatType& data,
+    arma::Row<size_t>& labels) const
+{
+  arma::mat scores;
+  Classify(data, labels, scores);
+}
+
+template <typename MatType, typename KernelType>
+void KernelSVM<MatType, KernelType>::Classify(
+    const MatType& data,
+    arma::Row<size_t>& labels,
+    arma::mat& scores) const
+{
+  Classify(data, scores);
+  double threshold = arma::as_scalar(arma::mean(scores, 1));
+
+  // Prepare necessary data.
+  labels.zeros(data.n_cols);
+
+  for (size_t i = 0; i< scores.n_elem; i++)
+  {
+    if (scores(i) >= threshold)
+      labels(i) = 1;
+    if (scores(i) < threshold)
+      labels(i) = 0;
+  }
+}
+
+template <typename MatType, typename KernelType>
+void KernelSVM<MatType, KernelType>::Classify(
+    const MatType& data,
+    arma::mat& scores) const
+{
+  if (fitIntercept)
+  {
+    scores = (data.t() * parameters.t()).t();
```

Comment:
Hey @saksham189  Can you help me understand. How can I calculate scores when using any kernel? I think we should do `kenel.Evaluate(x, y)` before multiplying with `parameters` but can't able to decide what should be the of value `y`. I tried multiple ways but didn't got better results.

Reply:
can you provide me a link to the implementation that you are referring?

Reply:
@saksham189  You can see this implementation https://github.com/kenextra/SVMAlgorithm/blob/master/SVM.py pretty much similar to our implementation :grin:

Reply:
alright so can you follow the implementation to create the scores?

Reply:
Yup, already did that's why builds are passing now :+1: